R/tt_sort.R

```diff
@@ -156,7 +180,16 @@ sort_at_path <- function(tt,
   }
 
   ## XXX hacky fix this!!!
+  ## tt_at_path removes root even if actual root table isn't named root, we need to match that behavior
+  if(path[1] == "root") {
```

Comment:
I just fixed a problem with `as_result_df` for cases with `rbind`ed tables, and seems that there is a `"rbind_root"` at the beginning of the path. I removed all initial "root" and "rbind_root" from there when doing as_result_df (also because it is not really informative), but I was wondering if we need to check for this also here or if we should fix it in the `rbind` function

Reply:
I'm not so sure. As I said, I actually think this behavior is outdated and should be deprecated (when the first element, whether root or rbind_root) doesn't actually match the path (ie there is no "root"/"rbind_root" subtable at the root level, its named something else). 

For when it DOES match, it already doesn't matter whether its there or not; you don't have to name the root element when specifying paths (I'm not sure if this is true everywhere... someone should check!, but its true of sort_at_path and tt_at_path, and I think the latter is used by a lot of the user-facing pathing support).

Put simply, I think we should be moving away from supporting users putting in paths that are *actively wrong*, since we now after this PR will give them a really nice error message that ells them how to figure out what paths are valid.

We (ie you guys) could also write a check_path_valid function that could be called at the top of things with pathing support, I suppose, but I'm not sure its worth a full additional walk of the tree, tbh.